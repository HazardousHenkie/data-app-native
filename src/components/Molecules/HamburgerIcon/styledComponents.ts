import styled from 'styled-components'

import { IconButton } from 'react-native-paper'
import MapView from 'react-native-maps'

const MapViewStyled = styled(MapView)`
    width: 100%;
    height: 100%;
`

export const IconButtonStyled = styled(IconButton)`
    padding-left: 18px;
    margin-top: 40px;
`

export default MapViewStyled
