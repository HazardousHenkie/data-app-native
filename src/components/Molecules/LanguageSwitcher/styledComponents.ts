import styled from 'styled-components'

import { View } from 'react-native'

const LanguageSwitcherWrapper = styled(View)`
    position: relative;
    width: 100%;
`

export default LanguageSwitcherWrapper
