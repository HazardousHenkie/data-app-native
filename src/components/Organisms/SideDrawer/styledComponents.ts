import styled from 'styled-components'

import { IconButton } from 'react-native-paper'

// export const DrawerChildren = styled.div`
//     width: 250px;
//     height: 100%;
// `

const IconButtonStyled = styled(IconButton)`
    padding-left: 18px;
    margin-top: 40px;
`

export default IconButtonStyled
