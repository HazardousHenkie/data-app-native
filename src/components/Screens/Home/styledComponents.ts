import styled from 'styled-components'

import MapView from 'react-native-maps'

const MapViewStyled = styled(MapView)`
    width: 100%;
    height: 100%;
`
export default MapViewStyled
