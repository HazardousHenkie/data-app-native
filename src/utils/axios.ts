import axios from 'axios'

const Axios = axios.create({
    baseURL: process.env.REACT_APP_API_URL,
})

Axios.defaults.headers.common['Content-Type'] = 'application/json;charset=utf-8'

export default Axios
