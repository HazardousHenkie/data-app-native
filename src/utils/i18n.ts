import i18n from 'i18next'
import * as Localization from 'expo-localization'

import { initReactI18next } from 'react-i18next'

import translationEN from '../../translations/locales/en.json'
import translationJA from '../../translations/locales/ja.json'
import translationNL from '../../translations/locales/nl.json'

const resources = {
    en: translationEN,
    ja: translationJA,
    nl: translationNL,
}

export const appLocales = ['en', 'ja', 'nl']

i18n.use(initReactI18next).init({
    resources,
    debug: true,
    lng: Localization.locale,
    fallbackLng: 'en',
    whitelist: appLocales,
    interpolation: {
        escapeValue: false,
    },
    react: {
        useSuspense: false,
        wait: true,
    },
})

export default i18n
