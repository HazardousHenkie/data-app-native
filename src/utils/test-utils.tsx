import React from 'react'
import { render } from '@testing-library/react-native'

// import { Provider } from 'react-redux'

import { ThemeProvider } from 'styled-components'
import { Provider as PaperProvider } from 'react-native-paper'

import { I18nextProvider } from 'react-i18next'

import { NavigationContainer } from '@react-navigation/native'
import i18n from './i18n'

import darkTheme from '../styles/themeStyles'
import variables from '../styles/variables'

// import configureStore from '../store/configureStore'

// TODO: add store back to test when there is a reducer

const customRender = (
    ui: React.ReactElement,
    {
        // initialState = {},
        // store = configureStore(initialState),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            // <Provider store={store}>
            <NavigationContainer>
                <I18nextProvider i18n={i18n}>
                    <ThemeProvider theme={variables}>
                        <PaperProvider theme={darkTheme}>
                            {children}
                        </PaperProvider>
                    </ThemeProvider>
                </I18nextProvider>
            </NavigationContainer>
            // </Provider>
        )
    }

    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions,
    })
}

export * from '@testing-library/react-native'

export { customRender as render }
