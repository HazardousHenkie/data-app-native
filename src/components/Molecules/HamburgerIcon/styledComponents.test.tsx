import React from 'react'

import { IconButtonStyled } from './styledComponents'

import 'jest-styled-components'
import { render } from '../../../utils/test-utils'

describe('<IconButtonStyled />', () => {
    it('should render and match the snapshot', () => {
        const { toJSON } = render(<IconButtonStyled icon="menu" />)

        expect(toJSON).toMatchSnapshot()
    })

    // it('should render a <SVG> tag', () => {
    //     const { toJSON } = render(<IconButtonStyled icon="menu" />)

    //     //  expect(toJSON.firstElementChild?.tagName).toEqual('BUTTON')
    // })
})
