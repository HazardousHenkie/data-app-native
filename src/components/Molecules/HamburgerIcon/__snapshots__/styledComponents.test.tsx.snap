// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButtonStyled /> should render and match the snapshot 1`] = `[Function]`;
